app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,62 @@
+class CommentsController < ApplicationController
+
+  before_action :authenticate_user, except: [:index, :show]
+  before_action :find_comment, only: [:show, :edit, :update, :destroy]
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @post = Post.find(params[:post_id])
+    @comment.post = @post
+    @comment.user = current_user
+    respond_to do |format|
+      if @comment.save
+        CommentsMailer.notify_post_owner(@post, @comment.user.full_name).deliver_later
+        format.html {redirect_to post_path(@post)}
+        format.js {render :create_success}
+      else
+        format.html {render "/posts/show"}
+        format.js {render js: "alert('failure');"}
+      end
+    end
+  end
+
+  def show
+  end
+
+  def edit
+  end
```

Comment:
if you are not using show or edit remove it or add a comment as to why they are there